### DISCUSSION TITLE
Mlops Hackathon Team3 discussion/suggestions

### DISCUSSION THREAD

**Discussion by azeltov** [MicrosoftLearning/mslearn-mlops#8] [Ideas]
Suggestions:

Replace portal ui resource group create commands to cli:
https://microsoftlearning.github.io/mslearn-mlops/documentation/00-set-up.html

az group create --location <location_region> --name rg-dev-mlops

**Comment by djdean** [comment]
Possible transient error when creating the dataset, solution is to wait a few minutes and retry

**Reply by madiepev** [reply]
resolved (workflows replaced with tasks)

**Comment by djdean** [comment]
Use existing SP/RG/Workspace. For this, an example JSON is needed ,see below:

{
  "clientId": "<Client ID>",
  "clientSecret": "<Client Secret Value>",
  "subscriptionId": "<Subscription ID>",
  "tenantId": "<Tenant ID>",
  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
  "resourceManagerEndpointUrl": "https://management.azure.com/",
  "activeDirectoryGraphResourceId": "https://graph.windows.net/",
  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
  "galleryEndpointUrl": "https://gallery.azure.com/",
  "managementEndpointUrl": "https://management.core.windows.net/"
}

**Reply by madiepev** [reply]
resolved (included in challenge 3)

**Comment by djdean** [comment]
Local CLI auth - need to set correct subscription and login

**Reply by madiepev** [reply]
resolved (included in coach guide)

**Comment by djdean** [comment]
Clarify dataset vs datastore for training data

**Reply by madiepev** [reply]
resolved (included in tasks for challenge 2)

**Comment by djdean** [comment]
VSCode is not necessarily needed, a development environment is all that's really needed. We might want to mention this in the setup/pre work.

**Reply by madiepev** [reply]
resolved (not instructed to use)

**Comment by djdean** [comment]
The SP creation/secret content is repeated in challenge 2

**Reply by madiepev** [reply]
resolved (removed from setup)

**Comment by djdean** [comment]
Wording in challenge 4 is confusing. There's no "pipeline" created in the previous steps, just triggering a job, yet challenge 4 asks them to run the "pipeline"

**Reply by madiepev** [reply]
resolved

**Comment by djdean** [comment]
Have a part 2 OpenLab that focuses on model monitoring, data drift, etc

**Reply by madiepev** [reply]
backlog

**Comment by djdean** [comment]
Having AML API coverage as a pre-req would be helpful

**Comment by djdean** [comment]
Add a data scientist focused aspect

**Reply by madiepev** [reply]
resolved (new challenge 1)
